### PR TITLE
Trip sharing emails

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds sending emails to confirmed users added to a monitored trip as primary traveler, companion, or observer.

### CODE REVIEW

File: src/test/java/org/opentripplanner/middleware/models/MonitoredTripTest.java [hunk 1]
```diff
@@ -41,4 +47,81 @@ void initializeFromItineraryAndQueryParamsShouldNotModifyModes() {
...
+        }
+        assertEquals(expected.observers, tripUsers.observers);
+    }
+
+    private static Stream<Arguments> createGetAddedUsersCases() {
+        MobilityProfileLite primary = new MobilityProfileLite();
+        primary.userId = "primary-user-id";
+
+        MobilityProfileLite newPrimary = new MobilityProfileLite();
+        primary.userId = "new-primary-user-id";
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/models/MonitoredTripTest.java]: `newPrimary ` and not `primary`

**Review by br648 — APPROVED** [review]
All looks good to me. Just the one minor nit which didn't effect the unit tests anyway.

File: src/test/java/org/opentripplanner/middleware/models/MonitoredTripTest.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/models/MonitoredTripTest.java]: Good catch, thank you! (https://github.com/ibi-group/otp-middleware/pull/273/commits/02d74f9db32244b6de46ab1d8be8d9aad86b8708)

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM!

**Pull request merged**